Sources/SimpleCurrencyConverter/SimpleCurrencyConverter.swift

```diff
@@ -1,6 +1,140 @@
+        }
     }
+    
+    public func getExchangeRate(
+        baseCurrency: Currency,
+        targetCurrencies: [Currency],
+        completion: @escaping (Result<[Currency : Double], Error>) -> Void) {
+            Task {
+                let result = try await networkManager?.requestExchangeRate(
+                    apiKey: exchangeRatesAPIKey, url: exchangeRatesURL.get(), base: baseCurrency, target: targetCurrencies)
+                switch result {
+                case .success(let rates):
+                    completion(.success(rates))
+                case .failure(let error):
+                    completion(.failure(error))
+                default: ()
+                }
+            }
+        }
+    
+    public func convert(
+        amount: Double,
+        baseCurrency: Currency,
+        targetCurrency: Currency,
+        completion: @escaping (Result<Double, Error>) -> Void) {
+            Task {
+                let result = try await networkManager?.requestConvert(
+                    apiKey: exchangeRatesAPIKey, url: convertURL.get(), amount: amount, base: baseCurrency, target: targetCurrency)
+                switch result {
+                case .success(let result):
+                    completion(.success(result))
+                case .failure(let error):
+                    completion(.failure(error))
+                default: ()
+                }
+            }
+        }
+    
+    public func convert(
+        amount: Double,
+        baseCurrency: Currency,
+        targetCurrencies: [Currency],
+        completion: @escaping (Result<[Currency : Double], Error>) -> Void) {
+            Task {
+                let result = try await networkManager?.requestExchangeRate(
+                    apiKey: exchangeRatesAPIKey, url: exchangeRatesURL.get(), base: baseCurrency, target: targetCurrencies)
+                switch result {
+                case .success(let rates):
+                    var modifiedRates = rates
+                    for rate in modifiedRates {
+                        let value = rate.value * amount
+                        modifiedRates.updateValue(value, forKey: rate.key)
+                    }
+                    completion(.success(modifiedRates))
+                case .failure(let error):
+                    completion(.failure(error))
+                default: ()
+                }
+            }
+        }
 }
+
+private extension SimpleCurrencyConverter {
+    
+    var apiKeyFromPlistFile: String? {
+        guard let key = Bundle.main.infoDictionary?["ExchangeRatesAPIKey"] as? String else {
+            assertionFailure("❗ Failed to get exchangerates.io API key from info.plist file ❗")
+            return nil
+        }
+        
+        return key
+    }
+    
+    var exchangeRatesURL: (Result<URL, Error>) {
+        if let existingURL = URL(string: "https://api.apilayer.com/exchangerates_data/latest") {
+            return .success(existingURL)
```

Comment:
@ts-dzmitry-kurski Can you confirm that `return` is not needed here?
Same question for `return` below.

Reply:
@malayli without `return` there is an error `Reference to member 'success' cannot be resolved without a contextual type` and the same error for the failure case.

Reply:
Ok. Thanks.